File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/IdCtTSTInfo.kt
package com.jetbrains.signatureverifier.serialization

import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.ASN1Primitive
import org.bouncycastle.asn1.cms.ContentInfo

@Serializable
data class IdCtTSTInfo(
  override val contentType: TextualInfo,
  val content: TextualInfo
) : EncapContentInfo {

  constructor(contentInfo: ContentInfo) : this(
    TextualInfo.getInstance(contentInfo.contentType),
    TextualInfo.getInstance(contentInfo.content)
  )

  override fun toPrimitive(): ASN1Primitive =
    listOf(
      contentType.toPrimitive(),
      TaggedObjectInfo.getTaggedObjectWithMetaInfo(
        TaggedObjectMetaInfo(0, 1),
        content.toPrimitive()
      )
    ).toDLSequence()
}

File: net/JetBrains.Serialization/src/Dataholders/IdCtTSTInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Org.BouncyCastle.Asn1.Cms;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class IdCtTSTInfo : EncapContentInfo
{
  [JsonProperty("ContentType")] protected override TextualInfo ContentType { get; }

  [JsonProperty("Content")] public TextualInfo Content { get; }

  [JsonConstructor]
  public IdCtTSTInfo(TextualInfo contentType, TextualInfo content)
  {
    ContentType = contentType;
    Content = content;
  }

  public IdCtTSTInfo(ContentInfo contentInfo)
    : this(TextualInfo.GetInstance(contentInfo.ContentType), TextualInfo.GetInstance(contentInfo.Content))
  {
  }

  protected override Asn1Encodable GetContentPrimitive() => Content.ToPrimitive();
}